src/System.Net.Sockets/src/System/Net/Sockets/_MultipleConnectAsync.cs

```diff
@@ -491,4 +634,65 @@ protected override void OnFail(bool abortive)
             }
         }
     }
-}
\ No newline at end of file
+
+    internal sealed class MultipleSocketMultipleConnectAsync : MultipleConnectAsync
+    {
+        private readonly SocketType _socketType;
+        private readonly ProtocolType _protocolType;
+        private readonly bool _supportsIPv4;
+        private readonly bool _supportsIPv6;
+
+        protected override bool RequiresUserConnectAttempt { get { return false; } }
+
+        protected override Socket UserSocket
+        {
+            get
+            {
+                Debug.Fail("Should never get here");
+                throw new NotImplementedException();
+            }
+        }
+
+        public MultipleSocketMultipleConnectAsync(SocketType socketType, ProtocolType protocolType)
+        {
+            Debug.Assert(!SocketPal.SupportsMultipleConnectAttempts);
+
+            _socketType = socketType;
+            _protocolType = protocolType;
+            _supportsIPv4 = Socket.OSSupportsIPv4;
+            _supportsIPv6 = Socket.OSSupportsIPv6;
+        }
+
+        protected override IPAddress GetNextAddress(out Socket attemptSocket)
+        {
+            if (_supportsIPv4 || _supportsIPv6)
+            {
+                while (nextAddress < addressList.Length)
```

Comment:
I realize it's not introduced in this change, but I was confused by nextAddress being a field.  Just a naming thing; I'd have expected _nextAddress.  _addressList as well for that matter.